daemon/cmd/state.go

```diff
@@ -284,6 +284,7 @@ func (d *Daemon) regenerateRestoredEndpoints(state *endpointRestoreState) (resto
 		log.WithField(logfields.EndpointID, ep.ID).Info("Successfully restored endpoint. Scheduling regeneration")
 		go func(ep *endpoint.Endpoint, epRegenerated chan<- bool) {
 			if err := ep.RegenerateAfterRestore(); err != nil {
+				log.WithField(logfields.EndpointID, ep.ID).WithError(err).Debug("error regenerating during restore")
```

Comment:
Now that this is debug, and the callee no longer has any such error message, if there is an error during restore there is no corresponding log message. I didn't look _too_ closely but it didn't seem like this error will ever be printed, so this could make it difficult to pinpoint the cause of a problem during restore. Should this be `Warn()` level, perhaps wrapped in the extra `errors.Is(..., context.Cancelled)` check?

Reply:
`ep.RegenerateAfterRestore()` returns one of `fmt.Errorf("failed while regenerating endpoint")` or `nil`, so errors aren't bubbled up to the caller. This is due to `Endpoint.Regenerate()` returning a `<-chan bool` and is probably something we should change at some point. This overuse of channels makes the code rather fragile and inflexible to changes.

Would you rather have a `Warn()` behind a check on `aliveCtx` for the time being? Note that as of `Endpoint.Regenerate()`, the context passed down to compile/attach can be either `regenMetadata.ParentContext` or `e.aliveCtx`, so only trusting `aliveCtx` is not technically correct. We need the error value to be sure.

Reply:
Hrm. This is more nuanced than I'd like. It seems that all of the `restoreIdentity()` failures are related to `aliveCtx` so we can safely ignore those. Basically the endpoint is just deleted while we are restoring it, so it's a normal operation not worth logging. Then for the `Endpoint.Regenerate()` case, as you say the error is getting a bit hidden. I'm not sure there's much point in logging the error here after all.

It looks like the underlying regeneration error will be logged to the endpoint's log when it fails, which is at least _something_. It won't end up in the main log if the initial restore fails though. I'm not really sure the best way to handle this right now, other than maybe to have `endpoint.Regenerate()` also log the regeneration error since that function actually receives the underlying regeneration error via the EventQueue's `resChan`.

As for this line of code, I'm fine to leave this at Debug now. Thanks for humouring me on   how to improve this area.